src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,189 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * This implementation checks that properties files are sorted.
+ * It is using the binary order from String.compare()
+ *
+ * @author Thomas Senger
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    public static final String MSG_IO_EXCEPTION_KEY = "unable.open.cause";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    @Override
+    protected void processFiltered(File file, FileText fileText) {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            log(1, MSG_IO_EXCEPTION_KEY, file.getPath(),
+                    ex.getLocalizedMessage());
+        }
+
+        String previousProp = "";
+
+        final Enumeration keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            //if (previousProp.compareTo(propKey) > 0) {
+            if (String.CASE_INSENSITIVE_ORDER.compare(previousProp, propKey) > 0) {
+
+                final int lineNumber = getLineNumber(fileText, propKey);
+                // Number of occurrences is number of duplications + 1
+                log(lineNumber, MSG_KEY, propKey, previousProp);
+            }
+
+            previousProp = propKey;
+        }
+    }
+
+    /**
+     * Method returns line number the key is detected in the checked properties
+     * files first.
+     *
+     * @param fileText
+     *            {@link FileText} object contains the lines to process
+     * @param keyName
+     *            key name to look for
+     * @return line number of first occurrence. If no key found in properties
+     *         file, 1 is returned
+     */
+    private static int getLineNumber(FileText fileText, String keyName) {
+        final Pattern keyPattern = getKeyPattern(keyName);
+        int lineNumber = 1;
+        final Matcher matcher = keyPattern.matcher("");
+        for (int index = 0; index < fileText.size(); index++) {
+            final String line = fileText.get(index);
+            matcher.reset(line);
+            if (matcher.matches()) {
```

Comment:
How reliable is this at finding the line number?
What if my key was `a` and I had a bunch of values before `a=` that had `a` in it? Is it possible it won't pick the right line?
Can keys be multi-line like values?
If the same key is used twice, it will report the wrong line.
If this isn't reliable, we shouldn't use it as people will just complain of false line numbers.